cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/wizard/DeployProgress/DeployProgress.tsx

```diff
@@ -0,0 +1,154 @@
+// React imports
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+
+// Library imports
+import { CdsButton } from '@cds/react/button';
+import { LazyLog } from 'react-lazylog';
+import { WebSocketHook } from 'react-use-websocket/dist/lib/types';
+
+// App imports
+import { useWebsocketService, WsOperations } from '../../../services/Websocket.service';
+import { NavRoutes } from '../../../constants/NavRoutes.constants';
+import './DeployProgress.scss';
+
+export const LogTypes = {
+    LOG: 'log',
+    STATUS: 'status'
+};
+
+interface StatusMessage {
+    type: string,
+    data: StatusMessageData
+}
+
+interface StatusMessageData {
+    data: {
+        status: string,
+        currentPhase: string,
+        totalPhases: Array<string>,
+        successfulPhases: Array<string>
+    }
+}
+
+function DeployProgress() {
+    let websocketSvc: WebSocketHook = useWebsocketService();
+
+    const [statusMessageHistory, setStatusMessageHistory] = useState<Array<StatusMessage>>([]);
+    const [logMessageHistory, setLogMessageHistory] = useState<Array<string>>(['Displaying logs']);
+
+    // Send message to websocket to start streaming cluster creation logs and status
+    useEffect(
+        () => {
+            websocketSvc.sendJsonMessage({
+                operation: WsOperations.LOGS
+            });
+        }, [] // eslint-disable-line react-hooks/exhaustive-deps
+    );
+
+    // Processes each message by type ('log' or 'status') and routes to appropriate handlers/state
+    useEffect(() => {
+        const lastMessage: MessageEvent | null = websocketSvc.lastMessage;
+        const logData = (lastMessage) ? JSON.parse(lastMessage.data) : null;
+
+        if (logData && logData.type === LogTypes.LOG) {
+            const logLine = formatLog(logData);
+            setLogMessageHistory((prev) => prev.concat([logLine]));
+        } else if (logData && logData.type === LogTypes.STATUS) {
+            // Note: deployment status not yet being used
+            setStatusMessageHistory((prev) => [logData.data]);
+        }
+
+    }, [websocketSvc.lastMessage]);
+
+    console.log(`TODO: output statusMessageHistory as steps complete in UI ${statusMessageHistory}`);
+
+    // Formats a log line to include pre-pended log type (INFO, WARNING, ERROR)
+    const formatLog = (log: any): string => {
```

Comment:
Is it possible to use another type instead of any?

Reply:
Yes, will do that